REGRESSION(296272@main): WPT /webdriver/tests/classic/element_click/center_point.py

smfr opened the pull request:
#### 98369d3def89a531afa81ee6bae8a1df22fde4b6
<pre>
REGRESSION(296272@main): WPT /webdriver/tests/classic/element_click/center_point.py
<a href="https://bugs.webkit.org/show_bug.cgi?id=295711">https://bugs.webkit.org/show_bug.cgi?id=295711</a>
<a href="https://rdar.apple.com/155546487">rdar://155546487</a>

Reviewed by Devin Rousso and BJ Burg.

After 296272@main `Widget::convertFromRootView()` did point conversion in floating point,
preserving the (0.5, 0.5) center point of a 1x1 element. This affected the result
of `WebAutomationSessionProxy::computeElementLayout()` which computes the element center point;
this caused events, which uses rounded coords, to either miss the 1x1 element, or not match the
computed center point of elements with odd-sized dimensions.

Fix by flooring in `computeElementLayout()`, which matches the behavior specified for computing
an element&apos;s in-view center point at <a href="https://w3c.github.io/webdriver/#interactability.">https://w3c.github.io/webdriver/#interactability.</a>

* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::computeElementLayout):

Canonical link: <a href="https://commits.webkit.org/297303@main">https://commits.webkit.org/297303@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e01baf277d634c558cbb0b17f961f337558dec71

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/111225 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30891 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/21340 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/117256 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/61493 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/31572 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/39473 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84538 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/114172 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/25208 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/100133 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64984 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24550 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/61076 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94586 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/18341 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/120330 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/38274 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/28438 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93465 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/38650 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/96408 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93290 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23782 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38385 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/16151 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/34259 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/38163 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/43640 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37828 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/41161 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/39530 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

burg reviewed (approved):
Nice fix, thank you!

smfr committed:
REGRESSION(296272@main): WPT /webdriver/tests/classic/element_click/center_point.py
https://bugs.webkit.org/show_bug.cgi?id=295711
rdar://155546487

Reviewed by Devin Rousso and BJ Burg.

After 296272@main `Widget::convertFromRootView()` did point conversion in floating point,
preserving the (0.5, 0.5) center point of a 1x1 element. This affected the result
of `WebAutomationSessionProxy::computeElementLayout()` which computes the element center point;
this caused events, which uses rounded coords, to either miss the 1x1 element, or not match the
computed center point of elements with odd-sized dimensions.

Fix by flooring in `computeElementLayout()`, which matches the behavior specified for computing
an element's in-view center point at https://w3c.github.io/webdriver/#interactability.

* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::computeElementLayout):

Canonical link: https://commits.webkit.org/297303@main
